src/js/configuration_panel/button-listeners.js

```diff
@@ -141,25 +201,39 @@ https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
         });
 
         // Remove the old configuration from Chrome's localStorage.
-        chrome.storage.local.remove(["gamepadConfiguration"], function () {
-            // Write the new gamepadConfiguration in localStorage.
-            chrome.storage.local.set({ gamepadConfiguration: gamepadConfiguration }, that.toggleSaveButtonState);
+        chrome.storage.local.remove([configurationName], function () {
+            var configurationWrapper = {};
+            configurationWrapper[configurationName] = gamepadConfiguration;
+
+            // Save the new configuration.
+            chrome.storage.local.set(configurationWrapper, function () {
+                /**
+                 * Toggle (disable) the buttons if the data is stored as
+                 * "gamepadConfiguration" and not the unsaved changes.
+                 */
+                if (configurationName === "gamepadConfiguration") {
+                    that.toggleSaveAndDiscardButtons();
+                }
+            });
         });
     };
 
     /**
      *
-     * Toggle the "Save Changes" button when the input configuration is changed.
+     * Toggle the "Save Changes" and "Discard Changes" buttons when the input
+     * configuration is changed.
      *
      * @param {Object} that - The configurationPanel component.
      * @param {Object} saveChangesButton - The "Save Changes" button on the panel.
+     * @param {Object} discardButton - The "Discard Changes" button on the panel.
      *
      */
-    gamepad.configurationPanel.buttonListeners.toggleSaveButton = function (that, saveChangesButton) {
+    gamepad.configurationPanel.buttonListeners.toggleSaveAndDiscardButtons = function (that, saveChangesButton, discardButton) {
+        saveChangesButton = saveChangesButton[0];
+        discardButton = discardButton[0];
         chrome.storage.local.get(["gamepadConfiguration"], function (gamepadConfigurationWrapper) {
```

Comment:
What I had in mind was to use the model listener infrastructure rather than checking to see if the state is "dirty" (unsaved changes) yourself.  I do like that you are able to tell if they undo their unsaved changes and get back to a "clean" state.  Let me look at the docs for a bit and think about how we might get both without having to check each time.

Reply:
OK, I talked to our architect at some length about this, he suggested keeping a copy of the last state saved to storage and then using [`fluid.model.diff`](https://github.com/fluid-project/infusion/blob/132aa848c1c55f88dac7e8ebd996c67bca8d482e/src/framework/core/js/DataBinding.js#L1425) to compare the current model to what's represented by the form.   This would tell you whether the new model has unsaved changes or not.

Reply:
To make this work, you'd also need to make sure all form changes are relayed to the model.  We typically use [fluid-binder](https://github.com/fluid-project/fluid-binder) for this.